### PR TITLE
Add information about webhook communication

### PR DESCRIPTION
**Pull request by marcomorain**

### CODE REVIEW

File: jekyll/_cci2/webhooks.md [hunk 1]
```diff
@@ -35,6 +35,25 @@ Webhooks can be leveraged for various purposes. Some possible examples might inc
 - Trigger internal notification systems to alert people when workflows/jobs complete.
 - Build your own automation plugins and tools.
 
+## Communication Protocol
+
+A webhook is sent whenever an event occurs on the CircleCI platform.
+
+A webhook is sent using a HTTP POST, to the URL that was registered when the webhook was created. The body of the HTTP POST contains payload encoded using JSON.
+
+CircleCI expects that the server that response to a webhook will return a 2xx response code. If a non-2xx response is recevied, CircleCI will retry at a later time. If CircleCI does not recive a response to the webhook within a short period of time, we will assume that delivery has failed, and we will rery at a later time. The timeout period is currently 5 seconds, but is subject to change during the preview period. The exact details of the retry policy are not currently documented, and are subject to change during the preview period. Please get in touch with our team if you have feedback about timeouts and retries.
```
Comment (aengelberg) [inline jekyll/_cci2/webhooks.md:44]: small typo fixes
```suggestion
CircleCI expects that the server that responds to a webhook will return a 2xx response code. If a non-2xx response is received, CircleCI will retry at a later time. If CircleCI does not receive a response to the webhook within a short period of time, we will assume that delivery has failed, and we will retry at a later time. The timeout period is currently 5 seconds, but is subject to change during the preview period. The exact details of the retry policy are not currently documented, and are subject to change during the preview period. Please get in touch with our team if you have feedback about timeouts and retries.
```

File: jekyll/_cci2/webhooks.md [hunk 2]
```diff
@@ -35,6 +35,25 @@ Webhooks can be leveraged for various purposes. Some possible examples might inc
...
+
+### Headers
+
+A number of HTTP headers are set on webhooks, as detailed in the table below.
+
+|Header Name|Value|
+|-|-|
+|Content-Type|`application/json`|
+|User-Agent|A string indicating that the sender was CircleCI (`CircleCI-Webhook/1.0`). The value is subject to change during the preview period.|
+|Circleci-Event-Type|The name of the event type, (`workflow-completed`, `job-completed`, etc.)|
```
Comment (aengelberg) [inline jekyll/_cci2/webhooks.md:54]: Terminology nitpick, I think this should just say "event type" not "name of the event type"
```suggestion
|Circleci-Event-Type|The type of event, (`workflow-completed`, `job-completed`, etc.)|
```

File: jekyll/_cci2/webhooks.md [hunk 3]
```diff
@@ -35,6 +35,25 @@ Webhooks can be leveraged for various purposes. Some possible examples might inc
 - Trigger internal notification systems to alert people when workflows/jobs complete.
 - Build your own automation plugins and tools.
 
+## Communication Protocol
+
+A webhook is sent whenever an event occurs on the CircleCI platform.
+
+A webhook is sent using a HTTP POST, to the URL that was registered when the webhook was created. The body of the HTTP POST contains payload encoded using JSON.
```
Comment (aengelberg) [inline jekyll/_cci2/webhooks.md:42]: possible wordsmithing
```suggestion
A webhook is sent using an HTTP POST, to the URL that was registered when the webhook was created, with a body encoded using JSON.
```

**Review by aengelberg — APPROVED** [review]
Thanks for adding these. I just have some small spelling/grammar suggestions.

File: jekyll/_cci2/webhooks.md [hunk 4]
```diff
@@ -35,6 +35,27 @@ Webhooks can be leveraged for various purposes. Some possible examples might inc
...
 - Build your own automation plugins and tools.
 
+## Communication Protocol
+{: #communication-protocol }
+
+A webhook is sent whenever an event occurs on the CircleCI platform.
+
+A webhook is sent using an HTTP POST, to the URL that was registered when the webhook was created, with a body encoded using JSON.
+
+CircleCI expects that the server that responds to a webhook will return a 2xx response code. If a non-2xx response is received, CircleCI will retry at a later time. If CircleCI does not receive a response to the webhook within a short period of time, we will assume that delivery has failed, and we will retry at a later time. The timeout period is currently 5 seconds, but is subject to change during the preview period. The exact details of the retry policy are not currently documented, and are subject to change during the preview period. Please get in touch with our team if you have feedback about timeouts and retries.
```
Comment (rosieyohannan) [inline jekyll/_cci2/webhooks.md:45]: Can we put a link to how we want people to get in touch? Support or Discuss or somewhere else?

File: jekyll/_cci2/webhooks.md [hunk 5]
```diff
@@ -35,6 +35,28 @@ Webhooks can be leveraged for various purposes. Some possible examples might inc
...
+### Headers
+{: #headers }
+
+A number of HTTP headers are set on webhooks, as detailed in the table below.
+
+Header Name | Value
+--- | ---
+Content-Type | `application/json`
+User-Agent | A string indicating that the sender was CircleCI (`CircleCI-Webhook/1.0`). The value is subject to change during the preview period.
+Circleci-Event-Type | The type of event, (`workflow-completed`, `job-completed`, etc.)
```
Comment (rosieyohannan) [inline jekyll/_cci2/webhooks.md:56]:
```suggestion
Circleci-Event-Type | The type of event, (`workflow-completed`, `job-completed`, etc.)
```

File: jekyll/_cci2/webhooks.md (hunk 4 above)
Comment (marcomorain) [reply jekyll/_cci2/webhooks.md:45]: @TylerConlee do you know where this should link to?

Comment (marcomorain) [reply jekyll/_cci2/webhooks.md]: I've added a link to Canny.

**Pull request merged**